chapter.md

```diff
@@ -0,0 +1,781 @@
+}
+```
+
+**FIXME: @renvrant https://github.com/renvrant/formcontrol-freaks/pull/3#discussion_r106822417**
+
+These interfaces define a store
+that contains a single object representing a character,
+rather than using the character itself as the store.
+That way,
+if we want to add other top-level items in future,
+we won't need to reorganize existing material.
+For example,
+the next version of our application might have this structure:
+
+```ts
+export interface IForm {
+  character: ICharacter;        // character info
+  equipment: IEquipment;        // character's equipment
+}
+```
+
+Equally,
+if we want to add more forms,
+we won't need to reorganize the existing material,
+and we can re-use common actions that work for any form.
+
+One consequence of this decision is that
+every action should contain the path to the particular part of the state it applies to.
+Agile purists might say that we shouldn't add this until we need it,
+and if our application consisted of just a login form and a message of the day,
+they'd be right.
+However,
+we have enough experience with applications of this kind to know that we're going to,
+and adding that to our architecture from the start will save us re-work later.
+
+Having decided on the structure of our store,
+the next step is to define its initial state.
+We will do this by creating a literal object that conforms to the `ICharacter` interface,
+and another that conforms to the `IForm` interface:
+
+```ts
+const characterInitialState: ICharacter = {
+  name: undefined,
+  bioSummary: {
+    age: undefined,
+    size: undefined,
+    alignment: undefined,
+    race: undefined,
+  },
+  skills: []
+};
+
+const initialState: IForm = {
+  character: characterInitialState
+};
+```
+
+We could fold the definition of `characterInitialState` into `initialState`
+to create one large literal object,
+but again,
+since we're likely to add more state information
+(like the equipment the character is carrying),
+we have decided that `initialState` will only ever be
+a list of top-level objects conforming to separately-defined interfaces.
+
+> We have defined our initial state in a file of its own
+> to make it easier to find and update.
+> In a real application,
+> it would probably be created by some sort of factory
+> so that we could swap in something else for testing.
+> Note that in real life, it would *not* come from a database:
+> instead, it's the blank slate that would be updated from a database
+> as the application is being bootstrapped.
+> And yes,
+> that bootstrap process would be implemented as
+> a series of update actions.
+
+Another early decision is that we will define functions to create actions,
+and that each action will have two elements:
+a string called `type` (which Redux requires),
+and a sub-object called `payload` that will contain
+the path to the part of the store it applies to
+and whatever extra values are needed to update the state.
+Redux doesn't mandate this,
```

Comment:
I think maybe move this explanation upward a little? So say we've got a sub-object called a payload, which Redux doesn't mandate the value of. We've made the decision that our payload will contain a path an a value, etc etc... Your payload can have whatever, really, so that's worth articulating.